tests/ui/mismatched_types/closure-mismatch.next.stderr

```diff
@@ -0,0 +1,67 @@
+error[E0277]: the trait bound `{closure@$DIR/closure-mismatch.rs:12:9: 12:12}: Foo` is not satisfied
+  --> $DIR/closure-mismatch.rs:12:9
+   |
+LL |     baz(|_| ());
+   |     --- ^^^^^^ unsatisfied trait bound
+   |     |
+   |     required by a bound introduced by this call
+   |
+   = help: the trait `for<'a> FnOnce(&'a ())` is not implemented for closure `{closure@$DIR/closure-mismatch.rs:12:9: 12:12}`
+   = note: expected a closure with arguments `(&(),)`
+              found a closure with arguments `(&(),)`
```

Comment:
This kinda sucks, but it's a diagnostics bug b/c we're pretty sloppy with binders when adding this note.

Reply:
Should be improved by https://github.com/rust-lang/rust/pull/139515.